### DISCUSSION TITLE
How to install an npm global tool when using the Node feature

### DISCUSSION THREAD

**Discussion by sliekens** [devcontainers/community#62] [Q&A]
Hello,

I'm converting a dev container based on a Dockerfile to a dev container that uses Features.

For example, my Dockerfile contains this:

``` dockerfile

# [Choice] Node.js version: none, lts, 16, 14, 12, 10
ARG NODE_VERSION="none"
ENV NVM_DIR=/usr/local/share/nvm
ENV NVM_SYMLINK_CURRENT=true \
    PATH=${NVM_DIR}/current/bin:${PATH}
RUN bash /tmp/library-scripts/node-debian.sh "${NVM_DIR}" "${NODE_VERSION}" "${USERNAME}" \
    && apt-get clean -y && rm -rf /var/lib/apt/lists/*

RUN su ${USERNAME} -c "npm install -g docsify-cli" 2>&1

````

Moving forward, I want to use the Feature:

``` json
{
  "features": {
    "ghcr.io/devcontainers/features/node:1": {}
  }
}
```

Question: where can I put the `npm install -g` command? I can't leave it in my Dockerfile, because npm will not be available at that point.

Thanks!

**Comment by jcbhmr** [comment]
I think you can use one of the lifecycle hook commands: https://containers.dev/implementors/json_reference/#lifecycle-scripts I think the best spot to put it is in the updateContentCommand, but you could also use the postCreateCommand if you want.

```jsonc
// devcontainer.json
{
  "features": {
    "ghcr.io/devcontainers/features/node:1": {}
  },
  "updateContentCommand": "npm install -g docsify-cli"
  // OR
  "postCreateCommand": "npm install -g docsify-cli"
}
```

example:

```jsonc
    "image": "mcr.microsoft.com/vscode/devcontainers/javascript-node:0-18",
    "features": {
        "ghcr.io/devcontainers/features/docker-in-docker:2": {}
    },
    "postCreateCommand": "npm install -g @devcontainers/cli",
```
https://github.com/devcontainers/features/blob/main/.devcontainer.json#L11-L14
